### ISSUE TITLE
Free-URL-Mode: alias can mask the ID of a Note.

### ISSUE THREAD

**Issue by ccoenen** [hedgedoc/hedgedoc#101]
I ran into a weird situation just now. I was not able to figure out the source of the problem, but here is the symptom:

Something or someone created an article with the Note ID but as an alias. This made accessing the actual document impossible. An example:

The real document: https://example.com/KVlJ9-dBQWKrMYWnq3Z1BA
Somehow another document with this url (alias/free url!): https://example.com/kvlj9-dbqwkrmywnq3z1ba

Whenever you would follow the first link, this code here would find an alias (because this match apparently is case-insensitive?) ...

https://github.com/codimd/server/blob/6462968e84e8d92292dd23764a9e558d7800147d/lib/models/note.js#L143-L151

... and then would redirect you to the "correct" spelling of the faulty alias.

https://github.com/codimd/server/blob/6462968e84e8d92292dd23764a9e558d7800147d/lib/response.js#L184

**Comment by ErikMichelson** [comment]
Maybe this could be fixed in 1.x with the next fix release?

**Comment by davidmehren** [comment]
Hmm, as far as I understand, the `where` in `findOne` is case-sensitive, so we should not find the alias `kvlj9-dbqwkrmywnq3z1ba` for the note id `KVlJ9-dBQWKrMYWnq3Z1BA`.

While testing, I noticed that browsers like to autocomplete URLs. In my case, I created a standard note (with a random id using upper and lower-case letters) first and then another note with the same id but all lower-case. Firefox then liked to complete the lowercase alias with the mixed-case id.
